src/typings.ts

```diff
@@ -420,17 +420,46 @@ export interface Signature {
   concatSig: string
 }
 
-export interface IdentityWalletSchema {
+// WALLET
+export type WalletTypeEthers = 'WalletTypeEthers'
+export type WalletTypeIdentity = 'WalletTypeIdentity'
+export type WalletType = WalletTypeEthers | WalletTypeIdentity
+
+export interface TLWalletSchema {
   TLWalletVersion: number
-  ethersKeystore: string
+  walletType: WalletType
+  meta?: any
+}
+
+interface EthersWalletMeta {
+  walletFromEthers: ethers.Wallet
```

Comment:
I thought we aggreed on saving the SigningKey, which consists of the mnemonic, the derivation path and the private key, now?

Reply:
I would not like to expose a type which is not from us, this breaks abstraction.

Reply:
Misunderstood something. Will changes this